File: javascript/raindrops/raindrops.js
export const convert = num => {
  const factors = [...Array(num).keys()]
    .map(n => n + 1)
    .reduce((factors, candidate) => {
      if (num % candidate === 0) factors.push(candidate);
      return factors;
    }, [])

  let sound = '';
  if (factors.includes(3)) sound += 'Pling';
  if (factors.includes(5)) sound += 'Plang';
  if (factors.includes(7)) sound += 'Plong';
  return sound || String(num);
};

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {
    fun convert(n: Int): String = buildString {
        if (n % 3 == 0) append("Pling")
        if (n % 5 == 0) append("Plang")
        if (n % 7 == 0) append("Plong")

        return if (isEmpty()) n.toString() else toString()
    }
}
